### PR TITLE
Adding RFduino board support

### PR DESCRIPTION
**Pull request by rfswarm**
I am sorry - i searched for some minutes but I just can not find this "flag" function. hm maybe someone can enlighten me.

### CODE REVIEW

File: boards/rfduino/board.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * @}
+ */
+
+#include "board.h"
+#include "cpu.h"
+
+void board_init(void)
+{
+    /* setup led(s) for debugging */
+    NRF_GPIO->PIN_CNF[LED_RED_PIN] = GPIO_PIN_CNF_DIR_Output;
```
Comment (d00616) [inline boards/rfduino/board.c]: Is there an red LED available? I think this line can be removed.

File: boards/rfduino/include/board.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * @name Assign the hardware timer
+ */
+#define HW_TIMER            TIMER_0
+
+/**
+ * @name Macros for controlling the on-board LEDs.
+ * @{
+ */
+#define LED_RED_PIN         8
```
Comment (d00616) [inline boards/rfduino/include/board.h]: Red LED is set but there is no toggle command.

**Comment by rfswarm** [comment]
The full rfduino support still needs some testing. the board is stm32/jlink flashing and booting with console ok. but other peripheral functionlity does not work yet -> so lets call it alpha version.

File: boards/rfduino/board.c [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * @ingroup     board_rfduino-nrf51822
+ * @{
+ *
+ * @file        board.c
+ * @brief       Board specific implementations for the RFduino NRF51822 board
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de> 
+ * @author      Jan Wagner
```
Comment (thomaseichinger) [inline boards/rfduino/board.c]: Please add an email address.

File: boards/rfduino/include/periph_conf.h [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * @name UART configuration
+ * @{
+ */
+#define UART_NUMOF          (1U)
+#define UART_0_EN           1
+#define UART_IRQ_PRIO       1
+
+/* UART pin configuration */
+#define UART_PIN_RX       0
```
Comment (haukepetersen) [inline boards/rfduino/include/periph_conf.h]: indention...

File: boards/rfduino/dist/openocd.cfg [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# nRF51822 Target
+source [find interface/stlink-v2.cfg]
+
+transport select hla_swd
+
+set WORKAREASIZE 0x4000
+source [find target/nrf51.cfg]
+
+# use hardware reset, connect under reset
+#reset_config srst_only srst_nogate
```
Comment (thomaseichinger) [inline boards/rfduino/dist/openocd.cfg]: Please remove if not needed.

File: boards/rfduino/include/board.h [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+ * @defgroup    board_rfduino-nrf51822 RFduino NRF51822
+ * @ingroup     boards
+ * @brief       Board specific files for the RFduino RF51822 board
+ * @{
+ *
+ * @file
+ * @brief       Board specific definitions for the RFduino RF51822 board
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Jan Wagner
```
Comment (mehlis) [inline boards/rfduino/include/board.h]: add mail address

File: boards/rfduino/dist/gdb.cfg [hunk 7]
```diff
@@ -0,0 +1 @@
+target extended-remote 127.0.0.1:3333
```
Comment (mehlis) [inline boards/rfduino/dist/gdb.cfg]: write ":3333"

File: boards/rfduino/Makefile.features [hunk 8]
```diff
@@ -0,0 +1,2 @@
+FEATURES_PROVIDED += cpp
+FEATURES_PROVIDED += periph_uart periph_gpio periph_random periph_rtt periph_cpuid
```
Comment (mehlis) [inline boards/rfduino/Makefile.features]: you don't implement all this features, so adapt this list to your implementation

File: boards/rfduino/include/board.h (hunk 6 above)
Comment (thomaseichinger) [inline boards/rfduino/include/board.h]: Mail please.

**Comment by haukepetersen** [comment]
Code looks good, I just don't have a board for testing... Did you create a wiki page for this board?

**Comment by mehlis** [comment]
@rfswarm you can add the board to the RIOT wiki, it's writable for everyone

**Comment by rfswarm** [comment]
hm I dont think this board is ready yet. please check the  "mapping" problems and test tool:
https://github.com/rfswarm/nrf51_mapping.

**Comment by rfswarm** [comment]
@mehlis I currenlty dont have a rfduino with me from what i can make a picture with (old wiki - with misssing pic) - i am not sure if we need pictures for swd - 3 wires - to rfduino.

https://github.com/rfswarm/Board-RFduino/wiki/Board:-RFduino

if ok tell me and ill gutenberg it to the riot wiki

**Comment by rfswarm** [comment]
Hm - i dont get the random periph running - i dont have more time to work on it.
Examples default works so I think the board should be added.

**Comment by mehlis** [comment]
@rfswarm it's totally ok to not implement optional features like this random interface

File: boards/rfduino/Makefile.features (hunk 8 above)
Comment (mehlis) [inline boards/rfduino/Makefile.features]: remove periph random in case it's not working

**Comment by mehlis** [comment]
@rfswarm for the wiki: is there any board datasheet to show the wirering on the board?

**Comment by rfswarm** [comment]
http://www.rfduino.com/wp-content/uploads/2014/03/rfd22102_r400_b002_sch.pdf - maybe?

I thought that the labeled pins on the board are ok and give enougth informaion.

btw. I can not take a photo of RFD22102 (classic rfduino board with pin header) any more because I
desolderd RFD22301 - it works but does not look beautyfull:)

File: boards/rfduino/Makefile.features (hunk 8 above)
Comment (rfswarm) [inline boards/rfduino/Makefile.features]: Hm i dont have any edit rights anymore. I think moved the cloned riot-os from my own account. Hm well anyone with edit right feel free to edit :) " from unknown repository"

Comment (thomaseichinger) [inline boards/rfduino/Makefile.features]: @rfswarm Where/To whom did you move it? I don't have any rights for "unknown repository" neither. :smile:

Comment (rfswarm) [inline boards/rfduino/Makefile.features]: sh.. I think I removed and added the repository at my own git account. puh i am trying to fix it - maybe I can remove the 2312 pull and add a new one (with the same content and changes). I hope to google a solution today.

Comment (rfswarm) [inline boards/rfduino/Makefile.features]: https://github.com/isaacs/github/issues/168

yes this is my probem. i am new to this github thing. well

@thomaseichinger 

thomas, I ll clone a new riot soon, copy the files and changes from this pull - and create a new pull ? - is this ok?

**Comment by thomaseichinger** [comment]
@rfswarm Any news on a new branch/PR? Should we go with this one?

**Comment by OlegHahm** [comment]
What's the latest here?

**Comment by rfswarm** [comment]
no news from my side. I did not care about it since then because the rfduino was only my first step to code nrf. so i switched to yunja board and haven't tested rfduino ever since. I think the wiki page is ok.
I need to copy this PR to a new PR because of my mentioned git f-up - if this if required I ll do it soon and re-pull.

**Comment by thomaseichinger** [comment]
@rfswarm did find time for this?

**Comment by thomaseichinger** [comment]
@rfswarm ping.

**Comment by kYc0o** [comment]
What's the status of this PR? it is really necessary to have a rfduino to ACK it?

**Comment by PeterKietzmann** [comment]
@kYc0o sure! AFAIK no one even tested this PR (because the lack of hardware) and as @rfswarm  stated  (already in may 2015): "*I did not care about it since then because the rfduino was only my first step to code nrf*" .
@rfswarm, @riotrf are you available?

**Comment by OlegHahm** [comment]
Closed as memo. Please shout or re-open if you disagree.

**Comment by kYc0o** [comment]
So, we won't try to get this hardware? I heard that some people at INRIA Lille will deploy a testbed using similar boards so maybe it's interesting to take a look into this ones...

**Comment by OlegHahm** [comment]
RFduino or Zigduino?

**Comment by kYc0o** [comment]
Oh yeah... you're right. I just look at the RFduino hardware and it has BLE features in a very small M0+. Maybe to support a BLE solution it could be interesting, but only if we can get the hardware ;)

**Comment by cladmi** [comment]
The board still has its page on the wiki: https://github.com/RIOT-OS/RIOT/wiki/Board%3A-RFduino

Should it be deleted ? Or at least point to this closed PR ?

**Pull request closed without merging**